tests/convert/test_convert.py

```diff
@@ -1,43 +1,89 @@
+OUTPUT_FPS = CONFIG["CONVERT"]["OUTPUT_FPS"]
+FPS_FROM_FILENAME = CONFIG["CONVERT"]["FPS_FROM_FILENAME"]
+DELETE_INPUT = CONFIG["CONVERT"]["DELETE_INPUT"]
+OVERWRITE = CONFIG["CONVERT"]["OVERWRITE"]
+
+T = TypeVar("T")
+YieldFixture = Generator[T, None, None]
+
+
+@pytest.fixture
+def test_convert_dir(test_data_dir: Path) -> Path:
+    return test_data_dir / "convert"
+
+
+@pytest.fixture
+def test_convert_tmp_dir(
+    test_data_tmp_dir: Path, test_convert_dir: Path
+) -> YieldFixture[Path]:
+    # Create tmp dir
+    test_convert_tmp_dir = test_data_tmp_dir / "convert"
+    test_convert_tmp_dir.mkdir(exist_ok=True)
+    # Copy test files to tmp dir
+    shutil.copytree(test_convert_dir, test_convert_tmp_dir, dirs_exist_ok=True)
+    # Delete video files from tmp dir (we create them during the tests)
+    extension = CONFIG["DEFAULT_FILETYPE"]["VID"]
+    video_files_to_delete = test_convert_tmp_dir.rglob(f"*{extension}")
+    for f in video_files_to_delete:
+        f.unlink()
+    # Yield tmp dir
+    yield test_convert_tmp_dir
+    # Teardown tmp dir after use in test
+    shutil.rmtree(test_convert_tmp_dir)
 
 
 def test_check_ffmpeg() -> None:
     """Tests if ffmpeg can be called as a subprocess"""
     check_ffmpeg()
 
 
-def test_convert(test_data_tmp_dir: Path, test_data_dir: Path) -> None:
+@pytest.mark.parametrize(
+    "test_case, output_filetype, input_fps, fps_from_filename",
+    [
+        ("default", OUTPUT_FILETYPE, INPUT_FPS, FPS_FROM_FILENAME),
+        ("fps_from_filename", OUTPUT_FILETYPE, 180, True),
+        ("input_fps_20", OUTPUT_FILETYPE, 20.0, False),
+        ("input_fps_40", OUTPUT_FILETYPE, 40.0, False),
+        ("output_filetype_avi", ".avi", INPUT_FPS, FPS_FROM_FILENAME),
+        ("output_filetype_mkv", ".mkv", INPUT_FPS, FPS_FROM_FILENAME),
+        ("output_filetype_mov", ".mov", INPUT_FPS, FPS_FROM_FILENAME),
+        ("output_filetype_mp4", ".mp4", INPUT_FPS, FPS_FROM_FILENAME),
+    ],
+)
+def test_pass_convert(
+    test_convert_dir: Path,
+    test_convert_tmp_dir: Path,
+    test_case: str,
+    output_filetype: str,
+    input_fps: float,
+    fps_from_filename: bool,
+) -> None:
     """Tests the main function of OTVision/convert/convert.py
     transforming short test videos from h264 to mp4 based on
     framerate specified as part of the file path using ffmpeg.exe
     """
 
-    # Get reference data
-    h264_ref_videos = get_files(paths=[test_data_dir], filetypes=[".h264"])
-    mp4_ref_videos = get_files(paths=[test_data_dir], filetypes=[".mp4"])
-
-    # Copy input data to temporary folder and start conversion
-    for h264_ref_video in h264_ref_videos:
-        shutil.copy2(
-            Path(h264_ref_video),
-            Path(h264_ref_video).parents[1]
-            / test_data_tmp_dir.name
-            / Path(h264_ref_video).name,
-        )
-    h264_test_videos = get_files(
-        paths=[test_data_tmp_dir],
-        filetypes=[".h264"],
+    # # Convert test h264 files
+    convert(
```

Comment:
Maybe use a local variable `test_case_dir` and `test_case_tmp_dir` and if used a lot throughout method?`

```python
test_case_convert_dir = test_convert_dir / test_case
test_case_convert_tmp_dir = test_convert_tmp_dir / test_case

def convert(
        paths=[test_case_convert_tmp_dir],
        output_filetype=output_filetype,
        input_fps=input_fps,
        fps_from_filename=fps_from_filename,
):
 # ...



```

Reply:
Of course! :)